src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -202,7 +202,12 @@ private BoundExpression VisitExpressionWithoutStackGuard(BoundExpression node)
                 case BoundKind.DelegateCreationExpression:
                     return VisitDelegateCreationExpression((BoundDelegateCreationExpression)node);
                 case BoundKind.FieldAccess:
-                    return VisitFieldAccess((BoundFieldAccess)node);
+                    var fieldAccess = (BoundFieldAccess)node;
+                    if (fieldAccess.FieldSymbol.IsCapturedFrame)
```

Comment:
The generated code can reference an inner frame (when the source references variables captured into that frame). This change also changes how the inner frame is referenced. I don't know whether that is a problem or not, but I think one would need to construct a test that contains an inner frame to find out. I don't think any of these new tests do.